src/coreclr/jit/instr.cpp

```diff
@@ -877,7 +900,16 @@ void CodeGen::inst_RV_TT(instruction ins,
                     regSet.verifyRegUsed(regTmp);
                     return;
             }
-#else  // !TARGET_ARM
+#else // !TARGET_ARM
+#ifdef TARGET_LOONGARCH64
+            if (emitter::isFloatReg(reg))
+                assert((ins == INS_fld_d) || (ins == INS_fld_s));
+            else if (emitter::isGeneralRegister(reg) && (ins != INS_lea))
+            { // TODO should amend for LOONGARCH64 !!!
+                // assert((ins==INS_ld_d) || (ins==INS_ld_w));
+                ins = size == EA_4BYTE ? INS_ld_w : INS_ld_d;
+            }
+#endif
```

Comment:
I'm wondering why this code is necessary, besides maybe the asserts?

That is, in the int register case, isn't the caller already passing the correct instruction?

Reply:
OK, I will add a comment.

For LoongArch64-ABI, the float arg might be passed by integer register when there is no float register left but there is integer register(s) left.

Thanks